src/main/software/amazon/event/ruler/NameState.java

```diff
@@ -175,12 +175,11 @@ boolean isEmpty() {
      * @param pattern The pattern used by the sub-rule to transition to this NameState.
      * @param isTerminal True indicates that the sub-rule is using pattern to match on the final event field.
      */
-    void addSubRule(final Object rule, final double subRuleId, final Patterns pattern, final boolean isTerminal) {
+    void addSubRule(final Object rule, final SubRuleContext subRuleId, final Patterns pattern, final boolean isTerminal) {
```

Comment:
wondering if here and elsewhere in the class if `rule` can be fetched from `subRuleId` instead

Reply:
This is possible, but looks like if we do that everywhere the patch will be much larger, at the same time I don't see much improvement from this change. Either option is fine with me.

Reply:
I think that's an acceptable next step for this PR, though digging into JMH variances would be better place to focus right now.